mautic/mautic
# Enhancement 19 3rd party sms gateways

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? |  No
| New feature? | Yes
| Automated tests included? | Yet to come
| Related user documentation PR URL |  None
| Related developer documentation PR URL |  Yet to come
| Issues addressed (#s or URLs) | https://github.com/mautic-internal/mautic-internal/issues/19
| BC breaks? | No
| Deprecations? | No

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR enables the integration of SMS transport into Mautic. Instead of directly calling Twilio, a transport chain is implemented. All services tagged **mautic.sms_transport** are registered into the chain and checked against integration to see if they're enabled.

It provides also a configuration in the Mautic settings allowing to select from enabled transports. Currently enabled transport should be stored as parameter **sms_transport** in local.php.

#### Steps to test this PR:
1. Verify that Twilio is used ad default service. Send and SMS or just verify log file failing on some Twilio error.
2. Verify configuration is stored in local parameters and it's still working :-)
